reference/mysqli/mysqli/construct.xml

```
@@ -137,79 +152,100 @@
-                    . mysqli_connect_error());
-        }
-    }
-}
+/* You should enable error reporting for mysqli before attempting to make a connection */
+mysqli_report(MYSQLI_REPORT_ERROR | MYSQLI_REPORT_STRICT);
 
-$db = new foo_mysqli('localhost', 'my_user', 'my_password', 'my_db');
+$mysqli = mysqli_connect('localhost', 'my_user', 'my_password', 'my_db');
 
-echo 'Success... ' . $db->host_info . "\n";
+/* Setting the proper charset after establishing a connection */
+mysqli_set_charset($mysqli, 'utf8mb4');
 
-$db->close();
-?>
+printf("Success... %s\n", mysqli_get_host_info($mysqli));
 ]]>
    </programlisting>
-   <para>&style.procedural;</para>
+   &examples.outputs.similar;
+   <screen>
+<![CDATA[
+Success... localhost via TCP/IP
+]]>
+   </screen>
+  </example>
+  <example>
+   <title>Extending mysqli class</title>
    <programlisting role="php">
 <![CDATA[
 <?php
-$link = mysqli_connect('localhost', 'my_user', 'my_password', 'my_db');
 
-if (!$link) {
-    die('Connect Error (' . mysqli_connect_errno() . ') '
-            . mysqli_connect_error());
+class MyMysqli extends mysqli {
+	public function __construct($host, $user, $pass, $db, $port, $socket, $charset) {
+		mysqli_report(MYSQLI_REPORT_ERROR | MYSQLI_REPORT_STRICT);
+		parent::__construct($host, $user, $pass, $db, $port, $socket);
+		$this->set_charset($charset);
+	}
 }
 
-echo 'Success... ' . mysqli_get_host_info($link) . "\n";
+$db = new MyMysqli('localhost', 'my_user', 'my_password', 'my_db', 3306, null, 'utf8mb4');
+]]>
+   </programlisting>
+  </example>
+  <example>
+   <title>Manual error handling</title>
+   <para>If error reporting is disabled, the developer is responsible for failure checking and their proper handling</para>
+   <para>&style.oop;</para>
+   <programlisting role="php">
+<![CDATA[
+<?php
 
-mysqli_close($link);
-?>
+$mysqli = @new mysqli('localhost', 'my_user', 'my_password', 'my_db');
+if ($mysqli->connect_errno) {
+	error_log($mysqli->connect_error);
+	exit;
+}
```

```suggestion
mysqli_reporting(MYSQLI_REPORT_OFF);
$mysqli = new mysqli('localhost', 'my_user', 'my_password', 'my_db');
if ($mysqli->connect_errno) {
	throw new RuntimeException('mysqli connection error: ' . $mysqli->connect_error);
}
```

1. Explicitly turn off error reporting instead of nudging people towards`@`. It also makes the example more similar to the others where `mysqli_reporting` is used.
2. Throw an exception instead of calling `exit;`, which avoids all sorts of problems down the line, especially with unit tests.

---

This is a good suggestion, but the idea here was to silence all errors including warnings. I think many people are used to this kind of error suppression, although I agree this is a terrible practice. 

If we are to make it more explicit, then do you mind if I prepend `error_reporting(0);` to the example? It will make it clear that PHP errors go hand in hand with mysqli errors.

---

Yes, that's a reasonable alternative.